src/components/3d/SimpleModelView.tsx

```diff
@@ -0,0 +1,148 @@
+import { padding, position } from '@rainbow-me/styles';
+
+export type ModelViewerProps = {
+  readonly setLoading: (loading: boolean) => void;
+  readonly loading: boolean;
+  readonly style?: ViewStyle;
+  readonly uri?: string;
+  readonly alt?: string;
+  readonly fallbackUri?: string;
+};
+
+const StyledWebView = styled(WebView)`
+  ${position.size('100%')};
+`;
+
+const ProgressIndicatorContainer = styled(Animated.View)`
+  ${position.size('100%')};
+  position: absolute;
+  align-items: flex-end;
+  justify-content: flex-end;
+  ${padding(10)};
+`;
+
+export default function ModelViewer({
+  loading,
+  setLoading,
+  style,
+  uri,
+  alt,
+  fallbackUri,
+}: ModelViewerProps): JSX.Element {
+  const opacity = React.useMemo(() => new Animated.Value(1), []);
+  const [visibility, setVisibility] = React.useState<boolean>(false);
+  const [progress, setProgress] = React.useState<number>(0);
+  const originWhiteList = React.useMemo(() => ['*'], []);
+  const onMessage = React.useCallback(
+    ({ nativeEvent: { data } }) => {
+      const { type, payload } = JSON.parse(data);
+      if (type === 'progress') {
+        setProgress(payload as number);
+      } else if (type === 'model-visibility') {
+        setVisibility(payload as boolean);
+      }
+    },
+    [setProgress, setVisibility]
+  );
+  const fallbackSource = React.useMemo(
+    () => ({
+      uri: fallbackUri,
+    }),
+    [fallbackUri]
+  );
+  const source = React.useMemo(
+    () => ({
+      html: `
+<!DOCTYPE html>
+<html lang="en">
+<head>
+  <meta charset="utf-8">
+  <script type="module" src="https://unpkg.com/@google/model-viewer@0.6.0/dist/model-viewer.js"></script>
+  <script nomodule src="https://unpkg.com/@google/model-viewer@0.6.0/dist/model-viewer-legacy.js"></script>
+  <title>GLBModelViewer</title>
+  <style>
+    * { margin:0; padding:0; }
+    html { height: 100%; width: 100%; }
+    body { height: 100%; width: 100%; }
+    model-viewer { width: 100%; height: 100% }
+  </style>
+</head>
+<body>
+  <model-viewer src=${JSON.stringify(uri)} alt=${JSON.stringify(
+        typeof alt === 'string' ? alt : ''
+      )} auto-rotate camera-controls autoplay shadow-intensity="1">
+    <div slot="progress-bar"></div>
+  </model-viewer>
+  <script type="text/javascript">
+    function shouldPostMessage(type, payload) {
+      window.ReactNativeWebView.postMessage(JSON.stringify({
+        type: type,
+        payload: payload,
+      }));
+    }
+    var modelViewer = document.querySelector("model-viewer");
+    modelViewer.addEventListener("progress", function (e) {
+      shouldPostMessage("progress", e.detail.totalProgress);
+    });
+    modelViewer.addEventListener("model-visibility", function (e) {
+      shouldPostMessage("model-visibility", !!e.detail.visible);
+    });
+  </script>
+</body>
+</html>
+    `.trim(),
+    }),
+    [alt, uri]
+  );
+  const didLoadModel = React.useMemo(() => !!visibility && progress === 1, [
```

Comment:
memoizing boolean (and so easily computable) thing probably gives worse performance than just calculating this one `&&` and `===` every time